From: Alexander Gerasiov <gq@debian.org>
Subject: Debian-specific changes in ldapscripts.conf and runtime
 In Debian we try to use some knowlege from other configs (see runtime.debian).
 So I commented out some values here, and add notes for Debian users.
 I've alo commented usage of some tools, suggested, but not really needed for
 common work.
 .
 Inclusion of runtime.debian added to runtime file.
Forwarded: not-needed

--- a/etc/ldapscripts.conf
+++ b/etc/ldapscripts.conf
@@ -16,19 +16,28 @@
 #  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
 #  USA.
 
+# Note for Debian users:
+# On Debian system ldapscripts will try to parse and use nslcd config.
+# Look on commented variables and description lines started with DEBIAN.
+# But you could override it's values here.
+
+
 # LDAP server
-SERVER="ldap://localhost"
+# DEBIAN: value from /etc/nslcd.conf (uri) is used.
+#SERVER="ldap://localhost"
 
 # Suffixes
-SUFFIX="dc=example,dc=com" # Global suffix
-GSUFFIX="ou=Groups"        # Groups ou (just under $SUFFIX)
-USUFFIX="ou=Users"         # Users ou (just under $SUFFIX)
-MSUFFIX="ou=Machines"      # Machines ou (just under $SUFFIX)
+# DEBIAN: values from /etc/nslcd.conf (base maps) are used.
+#SUFFIX="dc=example,dc=com" # Global suffix
+#GSUFFIX="ou=Groups"        # Groups ou (just under $SUFFIX)
+#USUFFIX="ou=Users"         # Users ou (just under $SUFFIX)
+#MSUFFIX="ou=Machines"      # Machines ou (just under $SUFFIX)
 
 # Authentication type
+# DEBIAN: value from /etc/nslcd.conf (sasl_mech) is used.
 # If empty, use simple authentication
 # Else, use the value as an SASL authentication mechanism
-SASLAUTH=""
+#SASLAUTH=""
 #SASLAUTH="GSSAPI"
 
 # Simple authentication parameters
@@ -60,11 +69,12 @@
 #GDUMMYMEMBER="uid=dummy,$USUFFIX,$SUFFIX"
 
 # User properties
-USHELL="/bin/sh"
-UHOMES="/home/%u"     # You may use %u for username here
+# DEBIAN: values from /etc/adduser.conf are used.
+#USHELL="/bin/sh"
+#UHOMES="/home/%u"     # You may use %u for username here
 CREATEHOMES="no"      # Create home directories and set rights ?
-HOMESKEL="/etc/skel"  # Directory where the skeleton files are located. Ignored if undefined or nonexistant.
-HOMEPERMS="700"       # Default permissions for home directories
+#HOMESKEL="/etc/skel"  # Directory where the skeleton files are located. Ignored if undefined or nonexistant.
+#HOMEPERMS="755"       # Default permissions for home directories
 
 # User passwords generation
 # Command-line used to generate a password for added users.
@@ -72,11 +82,12 @@
 # WARNING    !!!! This is evaluated, everything specified here will be run !
 # WARNING(2) !!!! Some systems (Linux) use a blocking /dev/random (waiting for enough entropy).
 #                 In this case, consider using /dev/urandom instead.
-PASSWORDGEN="cat /dev/random | LC_ALL=C tr -dc 'a-zA-Z0-9' | head -c8"
+#PASSWORDGEN="cat /dev/random | LC_ALL=C tr -dc 'a-zA-Z0-9' | head -c8"
 #PASSWORDGEN="pwgen"
 #PASSWORDGEN="echo changeme"
 #PASSWORDGEN="echo %u"
 #PASSWORDGEN="<ask>"
+PASSWORDGEN="pwgen"
 
 # User passwords recording
 # you can keep trace of generated passwords setting PASSWORDFILE and RECORDPASSWORDS
@@ -94,7 +105,7 @@
 SYSLOGLEVEL="info"
 
 # Temporary folder
-TMPDIR="/tmp"
+#TMPDIR="/tmp"
 
 # Various binaries used within the scripts
 # Warning : they also use uuencode, date, grep, sed, cut, which... 
@@ -123,12 +134,12 @@
 
 # Character set conversion : $ICONVCHAR <-> UTF-8
 # Comment ICONVBIN to disable UTF-8 conversion
-ICONVBIN="/usr/bin/iconv"
+#ICONVBIN="/usr/bin/iconv"
 #ICONVCHAR="ISO-8859-15"
 
 # Base64 decoding
 # Comment UUDECODEBIN to disable Base64 decoding
-UUDECODEBIN="/usr/bin/uudecode"
+#UUDECODEBIN="/usr/bin/uudecode"
 
 # Getent command to use - choose the ones used
 # on your system. Leave blank or comment for auto-guess.
--- a/lib/runtime
+++ b/lib/runtime
@@ -729,6 +729,9 @@
 
 ### Source configuration file
 
+_RUNTIME_DEBIAN="/usr/share/ldapscripts/runtime.debian"
+. "$_RUNTIME_DEBIAN" || end_die "Unable to source runtime.debian, exiting..."
+
 _CONFIGFILE=${LDAPSCRIPTS_CONF:="/etc/ldapscripts/ldapscripts.conf"}
 . "$_CONFIGFILE" || end_die "Unable to source configuration file ($_CONFIGFILE), exiting..."
 
